src/main/java/tc/oc/pgm/ghostsquadron/GhostSquadronMatchModule.java

```diff
@@ -0,0 +1,446 @@
+        if (xClose && zClose) {
+          mPlayer.sendWarning(
+              ChatColor.RED
+                  + AllTranslations.get()
+                      .translate("match.ghostSquadron.landmine.tooClose", player),
+              true);
+          return;
+        }
+      }
+
+      event.setCancelled(true);
+      this.landmines.put(place, mPlayer.getId());
+      this.landmineTeams.put(place, mPlayer.getCompetitor());
+
+      if (item.getAmount() > 1) {
+        item.setAmount(item.getAmount() - 1);
+      } else {
+        event.getPlayer().setItemInHand(null);
+      }
+
+      player.sendMessage(
+          ChatColor.GREEN
+              + AllTranslations.get().translate("ghostSquadron.landminePlanted", player));
+    }
+  }
+
+  @EventHandler
+  public void landmineExplode(final PlayerMoveEvent event) {
+    MatchPlayer player = this.getMatch().getPlayer(event.getPlayer());
+    if (!MatchPlayers.canInteract(player)) return;
+
+    Location to = event.getTo();
+    Iterator<Map.Entry<Location, UUID>> iterator = this.landmines.entrySet().iterator();
+
+    while (iterator.hasNext()) {
+      Map.Entry<Location, UUID> entry = iterator.next();
+      Location landmine = entry.getKey();
+      MatchPlayer placer = this.getMatch().getPlayer(entry.getValue());
+
+      if (placer == null || !placer.isParticipating()) {
+        iterator.remove();
+        continue;
+      }
+
+      Competitor placerTeam = this.landmineTeams.get(landmine);
+      if (placerTeam == player.getParty()) continue;
+
+      if (to.distanceSquared(landmine) < GhostSquadron.LANDMINE_ACTIVATION_DISTANCE_SQ) {
+        TNTPrimed tnt =
+            (TNTPrimed)
+                landmine
+                    .getWorld()
+                    .spawnEntity(landmine.clone().add(0, 1, 0), EntityType.PRIMED_TNT);
+        tnt.setFuseTicks(0);
+        tnt.setYield(1);
+
+        this.reveal(player.getBukkit());
+        getMatch().callEvent(new ExplosionPrimeByEntityEvent(tnt, placer.getBukkit()));
+        iterator.remove();
+        this.landmineTeams.remove(landmine);
+      }
+    }
+  }
+
+  /*
+   * SPIDER
+   */
+  public void spideySense(final MatchPlayer player) {
+    UUID userId = player.getId();
+
+    Date when = this.spideySenses.get(userId);
+    Date now = new Date();
+
+    if (when == null || now.getTime() > when.getTime() + GhostSquadron.SPIDEY_SENSE_COOLDOWN) {
+      this.spideySenses.put(userId, now);
+
+      player
+          .getBukkit()
+          .addPotionEffect(new PotionEffect(PotionEffectType.NIGHT_VISION, 4 * 20, 0), true);
+      player.getBukkit().addPotionEffect(new PotionEffect(PotionEffectType.SPEED, 4 * 20, 1), true);
+
+      player.getBukkit().playSound(player.getBukkit().getLocation(), Sound.SPIDER_IDLE, 5, 0);
+      player.getBukkit().playSound(player.getBukkit().getLocation(), Sound.SPIDER_IDLE, 5, 0.25f);
+      player.getBukkit().playSound(player.getBukkit().getLocation(), Sound.SPIDER_IDLE, 5, 0.5f);
```

Comment:
I'm not sure about the `Sound.SPIDER_IDLE`, it was `Sound.ENTITY_SPIDER_AMBIENT` in the original source.